spint_11

Ramilyushka opened the pull request:
1. Подключена зависимость  Package Dependencies
3. Реализован индикатор загрузки при авторизации

Ramilyushka committed:
1. Подключена зависимость  Package Dependencies
2. Реализован индикатор загрузки при авторизации

Ramilyushka committed:
Устранение гонки с POST запросом

Ramilyushka committed:
Блокировка UI при авторизации

Ramilyushka committed:
Добавлен сервис для получения базовой информации о пользователе
Запрос выполняется в SplashViewController
Информация о пользователе заполняется в ProfileViewController

Ramilyushka committed:
Запрос на получение ссылки для аватара ProfileImageService

Ramilyushka committed:
Заменить «старое» API для KVO на «новое» в WebViewViewController
Показать алерт с ошибкой в SplashViewController

Ramilyushka committed:
Загрузка аватарки профиля с помощью Kingfisher

Ramilyushka committed:
Заменила использование OAuth2TokenStorage на хранение и получение из хранилища токена с использованием SwiftKeychainWrapper

Ramilyushka committed:
Вёрстка кодом. Настраиваем UIWindow. Не работает, черный экран после лаунчскрина, не открывается SplashViewController

Ramilyushka committed:
Storyboard id для  ImagesListViewController

Ramilyushka committed:
Сверстан в коде SplashViewController

Ramilyushka committed:
Исправлен цвет фона в SplashViewController

DanilaG commented on ImageFeed/SplashView/SplashViewController.swift:
```diff
@@ -35,41 +57,89 @@ class SplashViewController: UIViewController {
+            message: "Не удалось войти в систему",
+            buttonText: "Ок",
+            completion: { _ in
+                OAuth2TokenStorage().removeToken()
```
Давай заменим на `let _ = OAuth2TokenStorage().removeToken()` или пометим функцию `removeToken()` `@discardableResult` чтобы xcode не показывал варнинги)

DanilaG commented on ImageFeed/Profile/ProfileImageService.swift:
```diff
@@ -0,0 +1,60 @@
+final class ProfileImageService {
+    
+    static let shared = ProfileImageService()
+    static let DidChangeNotification = Notification.Name(rawValue: "ProfileImageProviderDidChange")
```
Давай назовём константу с маленькой буквы, чтобы она по стилю соответствовала с названием других переменных в проекте?)

DanilaG commented on ImageFeed/SplashView/SplashViewController.swift:
```diff
@@ -35,41 +57,89 @@ class SplashViewController: UIViewController {
+            buttonText: "Ок",
+            completion: { _ in
+                OAuth2TokenStorage().removeToken()
+                exit(0)
```
Классно, что обрабатываешь действия на алерте и не даёшь пользователю "зависнуть" на экране, на котором ничего нельзя сделать. Но неожиданное завершение тоже пугает пользователей) Предлагаю показывать экран авторизации, после нажатия "Ок"

DanilaG commented on ImageFeed/Constants.swift:
```diff
@@ -8,10 +8,13 @@
 import Foundation
 
 enum Constants {
```
Классно, что константы вынесены в отдельный name space. Это повышает читабельность кода и упрощает работу с автодополнением)

DanilaG commented on ImageFeed/SplashView/SplashViewController.swift:
```diff
@@ -35,41 +57,89 @@ class SplashViewController: UIViewController {
+        let image = UIImage(named: "launch_screen") ?? UIImage()
+        
+        let imageView = UIImageView(image: image)
+        imageView.backgroundColor = .ypBlack
```
🔥, что цвета вынесены в отдельные константы) Это снижает вероятность допустить опечатки в их названии при использовании

Ramilyushka committed:
1) Исправила показ индикатора: теперь отображается на экране логина после авторизации
2) Исправила поведение алерта при сетевых ошибках: теперь по нажатию открывается экран авторизации и отображается на нужных контроллерах
3) Убрала проверки guard profile == nil else { return }, guard avatarImageURL == nil else { return } в сервисах профиля, так как блочились повторный запросы профиля после показа алерта с ошибкой и в целом кажется эта проверка не нужна. Также предполагаю, что это аффектило бы сценарий смены пользователя
4) Добавила обнуление lastCode в oAuthService так как блочился повторный запрос  токена после показа алерта с ошибкой
